c++/src/kj/async-io-test.c++

```diff
@@ -238,6 +238,49 @@ TEST(AsyncIo, CapabilityPipe) {
   EXPECT_EQ("foo", result2);
 }
 
+TEST(AsyncIo, CapabilityPipeBlockedSendStream) {
+  // Check for a bug that existed at one point where if a sendStream() call couldn't complete
+  // immediately, it would fail.
+
+  auto io = setupAsyncIo();
+
+  auto pipe = io.provider->newCapabilityPipe();
+
+  Promise<void> promise = nullptr;
+  Own<AsyncIoStream> endpoint1;
+  uint nonBlockedCount = 0;
+  for (;;) {
+    auto pipe2 = io.provider->newCapabilityPipe();
+    promise = pipe.ends[0]->sendStream(kj::mv(pipe2.ends[0]));
+    if (promise.poll(io.waitScope)) {
+      // Send completed immediately, because there was enough space in the stream.
+      ++nonBlockedCount;
+      promise.wait(io.waitScope);
+    } else {
+      // Send blocked! Let's continue with this promise then!
+      endpoint1 = kj::mv(pipe2.ends[1]);
+      break;
+    }
+  }
+
+  KJ_DBG(nonBlockedCount);
```

Comment:
Vestigial KJ_DBG

Reply:
oops, fixed. (It printed ~300 FWIW.)

Reply:
Or 278, to be precise. A weirdly non-round number, and smaller than I would have expected.

Reply:
Nice, I was curious. I was expecting it to be in the thousands.